src/adaptors/http1/http1_server.c

```diff
@@ -674,170 +948,93 @@ static void _handle_connection_events(pn_event_t *e, qd_server_t *qd_server, voi
         }
         return;
     }
-    case PN_RAW_CONNECTION_NEED_WRITE_BUFFERS: {
-        _send_request_message((_server_request_t*) DEQ_HEAD(hconn->requests));
-        break;
-    }
-    case PN_RAW_CONNECTION_NEED_READ_BUFFERS: {
-        _handle_conn_need_read_buffers(hconn);
-        break;
-    }
     case PN_RAW_CONNECTION_WAKE: {
         // Note: wake events may occur before the raw connection is established
         if (sys_atomic_set(&hconn->q2_restart, 0)) {
             // note: unit tests grep for this log!
             qd_log(log, QD_LOG_TRACE, "[C%"PRIu64"] server link unblocked from Q2 limit", hconn->conn_id);
             hconn->q2_blocked = false;
-            _handle_conn_need_read_buffers(hconn);  // restart receiver flow
+            _replenish_empty_read_buffers(hconn);  // restart receiver flow
         }
 
         // note that when qdr_connection_process() handles the connection close
         // the hconn->qdr_conn pointer will be zeroed.
         while (hconn->qdr_conn && qdr_connection_process(hconn->qdr_conn))
             ;
-
-        qd_log(log, QD_LOG_DEBUG, "[C%"PRIu64"] Connection processing complete", hconn->conn_id);
         break;
     }
-    case PN_RAW_CONNECTION_DRAIN_BUFFERS: {
-        pn_raw_connection_t *pn_raw_conn     = pn_event_raw_connection(e);
-        int                  drained_buffers = qd_raw_connection_drain_read_write_buffers(pn_raw_conn);
-        qd_log(log, QD_LOG_DEBUG, "[C%" PRIu64 "] PN_RAW_CONNECTION_DRAIN_BUFFERS Drained a total of %i buffers",
-               hconn->conn_id, drained_buffers);
-    } break;
+    case PN_RAW_CONNECTION_NEED_READ_BUFFERS: {
+        _replenish_empty_read_buffers(hconn);
+        break;
+    }
     case PN_RAW_CONNECTION_READ: {
-        int error = _handle_conn_read_event(hconn);
-        if (error)
-            qdr_http1_close_connection(hconn, "Incoming response message failed to parse");
+        // handled below in I/O loop
+        break;
+    }
+    case PN_RAW_CONNECTION_CLOSED_READ: {
+        hconn->q2_blocked = false;
+        // handled in I/O loop
+        break;
+    }
+    case PN_RAW_CONNECTION_NEED_WRITE_BUFFERS: {
+        // handled in I/O loop
         break;
     }
     case PN_RAW_CONNECTION_WRITTEN: {
```

Comment:
I think this is already accounted for in the latest patches. http1_server/client.c print every proton event passed to the handler. adaptor_tls.c now logs every octet read/written to both the TLS layer and the raw connection.  For example: 

https://github.com/kgiusti/skupper-router/blob/ISSUE-676-HTTP1-TLS/src/adaptors/adaptor_tls.c#L913

logs how many encrypted octets were taken from TLS and written to the raw connection

Reply:
@ganeshmurthy 
Here's actual log output. For unencrypted data flowing from the adaptor thru TLS to the raw connection:

`[C7] 257 unencrypted octets taken by TLS for encryption (1 buffers)`
`[C7] 279 encrypted octets written to the raw connection by TLS (1 buffers)`
`[C7] client proactor event PN_RAW_CONNECTION_WRITTEN`

And here's the reverse: encrypted coming in from the raw connection, going through TLS and up to the adaptor:

`[C7] client proactor event PN_RAW_CONNECTION_READ`
`[C7] 111 encrypted octets read from the raw connection passed to TLS for decryption (1 buffers)`
`[C7] 89 decrypted octets taken from TLS for adaptor input (1 buffers)`

Does that give us enough information on the octet flows?

Reply:
This looks good to me. Just a minor comment. In the `PN_RAW_CONNECTION_WRITTEN` event case, the log messages are written prior to the logging of the actual event and in the case of `PN_RAW_CONNECTION_READ` the log message are written after the logging of the actual event. Can we fix the `PN_RAW_CONNECTION_READ` event log message to appear *after* the actual octet counts so it looks uniform? Also change octets to bytes so it is easy for a pedestrian person looking at the logs to understand ?

Reply:
Changing the wording to bytes is fine.

I wouldn't recommend changing the order of the logs.

That log sequence mirrors exactly the sequence of events as they are processed by the code: you're going to get a READ event _first_, because that kicks off the I/O loop which (may) result in reading bytes from the raw connection. So the READ event will always come before any READ-related I/O is done. The reverse is true for WRITTEN: proactor won't issue a WRITTEN event until after the adaptor writes bytes to the raw connection and returns control back to proactor for it to do the socket write. The WRITTEN even actually posts later, after the socket write returns.

Having the trace log reorder events from their actual flow is going to misleading.